l10n_id_taxform_period/models/tax_period.py

```diff
@@ -0,0 +1,154 @@
+            strWarning = _("The start date must precede it's end date")
+            raise models.ValidationError(strWarning)
+
+    @api.multi
+    def action_create_period(self):
+        for year in self:
+            year._create_period()
+
+    @api.multi
+    def _create_period(self):
+        self.ensure_one()
+        obj_period = self.env["l10n_id.tax_period"]
+        date_start = datetime.strptime(self.date_start, "%Y-%m-%d")
+        while date_start.strftime("%Y-%m-%d") < self.date_end:
+            date_end = date_start + relativedelta(months=+1, days=-1)
+
+            if date_end.strftime("%Y-%m-%d") > self.date_end:
+                date_end = datetime.strptime(self.date_end, "%Y-%m-%d")
+
+            obj_period.create({
+                "name": date_start.strftime("%m/%Y"),
+                "code": date_start.strftime("%m/%Y"),
+                "date_start": date_start.strftime("%Y-%m-%d"),
+                "date_end": date_end.strftime("%Y-%m-%d"),
+                "year_id": self.id,
+            })
+            date_start = date_start + relativedelta(months=+1)
+
+    @api.model
+    def _find_year(self, dt=None):
+        if not dt:
+            dt = datetime.now().strftime("%Y-%m-%d")
+        criteria = [
+            ("date_start", "<=", dt),
+            ("date_end", ">=", dt),
+        ]
+        results = self.search(criteria)
+        if not results:
+            strWarning = _("No tax year configured for %s" % dt)
+            raise models.ValidationError(strWarning)
+        result = results[0]
+        return result
+
+
+class TaxPeriod(models.Model):
+    _name = "l10n_id.tax_period"
+    _description = "Tax Period"
+    _order = "date_start asc, id"
+
+    name = fields.Char(
+        string="Tax Period",
+        required=True,
+    )
+    code = fields.Char(
+        string="Code",
+        required=True,
+    )
+    date_start = fields.Date(
+        string="Date Start",
+        required=True,
+    )
+    date_end = fields.Date(
+        string="Date End",
+        required=True,
+    )
+    year_id = fields.Many2one(
+        string="Tax Year",
+        comodel_name="l10n_id.tax_year",
+        ondelete="cascade",
+    )
+
+    @api.constrains("date_start", "date_end")
+    def _check_range(self):
+        if self.date_end <= self.date_start:
+            strWarning = _("The start date must precede it's end date")
+            raise models.ValidationError(strWarning)
+
+    @api.multi
+    def _next_period(self, step):
+        self.ensure_one()
+        criteria = [
+            ("date_start", ">", self.date_start)
+        ]
+        results = self.search(criteria)
+        if results:
+            return results[step - 1]
+        return False
+
+    @api.multi
+    def _previous_period(self, step):
+        self.ensure_one()
+        criteria = [
+            ("date_start", "<", self.date_start)
+        ]
+        results = self.search(criteria, order="date_start desc")
```

Comment:
You are using: `results[step - 1]` to return the values.
Maybe you could use `self.search(... offset=step-1, limit=1)` parameter